### ISSUE TITLE
Error seu@assays$RNA$counts : $ operator not defined for this S4 class.  for doubletFinder( )

### ISSUE THREAD

**Issue by chingyaousf** [chris-mcginnis-ucsf/DoubletFinder#207]
Error seu@assays$RNA$counts : $ operator not defined for this S4 class. 
If the issue of compatibility since I couldn't upgrade Seurat package to ‘5.1.0’ in the server which has no issue of running?

> packageVersion("DoubletFinder")
[1] ‘2.0.4’
> packageVersion("Seurat")
[1] ‘4.3.0.1’
> # Find doublets using DoubletFinder
> suppressMessages(require(DoubletFinder))
> nExp <- round(ncol(seurat_object) * 0.04)  # expect 4% doublets
> seurat_object <- doubletFinder(seurat_object, pN = 0.25, pK = 0.09, nExp = nExp, PCs = 1:10)
Error in seu@assays$RNA$counts : $ operator not defined for this S4 class

**Comment by Hui-XiangZhang** [comment]
I have the same question when using Seurat 4.3 .
Who can sovle this?

**Comment by chingyaousf** [comment]
I attempted to solve this issue and it appears to have worked. For detailed instructions, please check the following link:
[Fix for DoubletFinder Error in seu@assays$RNA$counts Operator Not Defined for this S4 Class]
https://github.com/chingyaousf/fix--doubletFinder-Error-in-seu-assays-RNA-counts-operator-not-defined-for-this-S4-class

**Comment by ricardo-aaron** [comment]
Yes, changing seu@assays$RNA$counts to seu@assays$RNA@counts in doubletFinder.R AND paramSweep.R and reinstalling DoubletFinder works.

**Comment by 02-i-need-healing** [comment]
> Agreee with ricardo-aaron, it works! 

I have modified the corresponding code and compressed it into a zip file. You can save it to your folder and use the following command to install and load it:

`remotes::install_local("your path/DoubletFinder.zip")`

[DoubletFinder.zip](https://github.com/user-attachments/files/16679708/DoubletFinder.zip)

**Comment by chris-mcginnis-ucsf** [comment]
Has been fixed by @haibol2016
